service/src/test/java/org/mskcc/smile/service/SampleServiceTest.java

```diff
@@ -323,4 +323,43 @@ public void testFindSamplesByDate() throws Exception {
                 sampleRepository.findLatestSampleMetadataBySmileId(updatedSample.getSmileSampleId());
         Assertions.assertThat(latestMetadata.getCmoSampleName()).isEqualTo("C-LATESTLABEL-T02-d002");
     }
+
+    /**
+     * Tests that sample can be found by a valid igoId
+     * @throws Exception
+     */
+    @Test
+    public void testFindSampleByIgoId() throws Exception {
```

Comment:
I would add or expand these tests to make sure that searching for the same sample by whichever ID is returning the same instance of that sample instead of just verifying that the returned object is not null.